File: media-service/src/main/java/dev/videostreaming/microservice/mediaservice/Media.java
package dev.videostreaming.microservice.mediaservice;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.Instant;

@Data
@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "media")
public class Media {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    private String uploaderId;

    private String title;

    private String originalFileName;

    private String contentType;

    private BigDecimal fileSize;

    private String fileExtension;

    private String sourceBucketName;

    private String sourceObjectKey;

    private String processedObjectKey;

    @Enumerated(EnumType.STRING)
    private MediaStatus status;

    private BigDecimal duration;

    private BigDecimal width;

    private BigDecimal height;

    private String failureReason;

    private Instant createdAt;
    private Instant updatedAt;
}

File: frontend/src/types/media.ts
export type MediaStatus = 'UPLOADING' | 'UPLOADED' | 'QUEUED' | 'READY' | 'FAILED';

export interface Media {
    id: string;
    title: string | null;
    status: MediaStatus;
    playbackUrl: string | null;
    duration: number | null;
    width: number | null;
    height: number | null;
    createdAt: string;
    updatedAt: string;
}

export interface CreateUploadResponse {
    mediaId: string;
    presignedUrl: string;
    uploadStatus: string;
    createdAt: string;
    updatedAt: string;
}